include/rmm/mr/device/binning_memory_resource.hpp

```diff
@@ -99,9 +100,15 @@ class binning_memory_resource final : public device_memory_resource {
   binning_memory_resource& operator=(binning_memory_resource&&)      = delete;
 
   /**
-   * @brief Get the upstream memory_resource object.
-   *
-   * @return UpstreamResource* the upstream memory resource.
+   * @briefreturn{rmm::device_async_resource_ref to the upstream resource}
+   */
+  [[nodiscard]] rmm::device_async_resource_ref get_upstream_resource() const noexcept
+  {
+    return upstream_mr_;
```

Comment:
non-blocking question: I wonder if we should make the naming of upstream resources consistent. Seems about 50/50 `upstream_` and `upstream_mr`

Reply:
Oh yeah, that rubbed me too, but I thought that it would be a bit to far away from the purpose of this PR